### PR TITLE
First commit UI for CrashMonitoring

### PR DESCRIPTION
**Pull request by puneetg1983**
First commit for UI for Crash Monitoring. We are enabling this on demo subscriptions only to get feedback on the UI from various people and to be able to write a blog on this feature. 

image

image

image

image

### CODE REVIEW

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring.component.html [hunk 1]
```diff
@@ -0,0 +1,148 @@
+<div class="category-container" style="width:60%">
+  <p class="category-description">
+    If your app is experiencing crashes, enable Crash Monitoring to collect memory dumps and callstack information to
+    identify the root cause of the crashes.Starting or stopping the monitoring will restart the app.
```
Comment (yunjchoi) [inline AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring.component.html:4]: Adding a space:
"identify the root cause of the crashes**. S**tarting or stopping the monitoring will restart the app.

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                  <div *ngIf="insights.length === 0">Crash Monitoring session is in progress. Please return to this page
+                    to view data collected on future crashes.</div>
+                  <div *ngIf="insights.length > 0">Crash Monitoring session is in progress. See the insights below for previously collected data</div>
+                </div>
+                <div *ngIf="!monitoringEnabled">
+                  <i class="fa health-icon fa-check-circle healthy-icon-color" aria-hidden="true"></i>
+                  The crash monitoring session has finished.
+                  <span *ngIf="insights != null && insights.length > 0">Please see the insights below.</span>
+                  <span *ngIf="insights != null && insights.length === 0">
+                    No crashes were detected from your last Crash Monitoring session.</span>
```
Comment (yunjchoi) [inline AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html:28]: Do we have another message to show when the user returns to the tool more than 15 days later when the data is no longer available in the Analyze section, but they have the app setting enabled?

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+            <div *ngIf="crashMonitoringSettings == null">
+              Crash Monitoring is not enabled for this app
+            </div>
+            <div *ngIf="crashMonitoringSettings != null">
+              <div>
+                <div *ngIf="monitoringEnabled" class="data-collection-loading">
+                  <div class="dot-flashing"></div>
+                  <div *ngIf="insights.length === 0">Crash Monitoring session is in progress. Please return to this page
+                    to view data collected on future crashes.</div>
+                  <div *ngIf="insights.length > 0">Crash Monitoring session is in progress. See the insights below for previously collected data</div>
```
Comment (yunjchoi) [inline AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html:21]: Should we call out not to delete dumps while the tool is running?

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring.component.html [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+      </div>
+      <collapsible-list-item body>
+        <div class="container">
+          <div>
+            <div class="row mt-4">
+              <div class="col-md-3 field-label">
+                Storage account <span class="required">*</span>
+
+                <fab-tooltip-host
+                  content= "The selected storage account will store the memory dumps captured via Crash Monitoring. It is strongly advised that you use one storage account per app. Selecting a storage account already in use for another app may cause Crash Monitoring to fail"
```
Comment (yunjchoi) [inline AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring.component.html:28]: Do we also show this message in the storage selection context panel or just in the tool tip?

Comment (puneetg1983) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring.component.html:28]: I was thinking that the storage component that we created now may be reused in other places of Diagnose and solve in future. That way, we can have a generic storage selection view everywhere. Considering that, I think it's better not to call out feature wise warnings in Storage selection view but I can change it if you think otherwise.

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html (hunk 3 above)
Comment (puneetg1983) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html:21]: Yes we should. That's a good idea. Can you give me the exact phrase or text to call out?

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html (hunk 2 above)
Comment (puneetg1983) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html:28]: Whenever the user returns after the session has stopped and if no data has been collected, the above message will be displayed. Now whether he comes back after a few days or 15 days, the scenario is still the same. Sorry not understanding the point you are making

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring.component.html (hunk 1 above)
Comment (puneetg1983) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring.component.html]: done

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring.component.ts [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+  }
+
+  onSelectEndDateHandler(event: any) {
+    if (event != null && event.date != null) {
+      this.endDate = event.date;
+    }
+  }
+
+  startMonitoring() {
+    if (this.validateSettings()) {
```
Comment (cindylovescoding) [inline AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring.component.ts:182]: Do we want to add some logging for these actions?

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html [hunk 6]
```diff
@@ -0,0 +1,171 @@
+<collapsible-list-fabric [collapsed]="collapse[0].collapsed" [title]="collapse[0].title">
```
Comment (cindylovescoding) [inline AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html:1]: Thanks for taking care of accessibility for the new component!

**Comment by cindylovescoding** [comment]
Hey @puneetg1983 Somehow I feel like the "expand" and "collapse" icon in your screenshot should be swapped:

image

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html (hunk 2 above)
Comment (yunjchoi) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html:28]: The scenario I have in mind is when the customer viewed the crashes detected, checked them, and came back to the page more than 15 days later. In that case, we will no longer show the previously collected data, right? What kind of message should we show in the Analyze section in that case?

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html (hunk 3 above)
Comment (yunjchoi) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html:21]: For line 21, we could say, "Crash Monitoring session is in progress. See the insights below for collected data and do not delete data while the tool is running." though it's a bit wordy.

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring.component.html (hunk 4 above)
Comment (yunjchoi) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring.component.html:28]: Ok, I think that makes sense.

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html (hunk 2 above)
Comment (puneetg1983) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html:28]: The Analyze section will show data only of the current session. If after 15 days, the session has expired, then the user will see what is displayed above. If the session is still running, then they will see "Crash Monitoring session in progress".

The View History section will automatically update to include all the data collected. So if they click on View History, then can see all the data. Does that answer the question? I am sorry I am still not fully clear of the scenario you are referring to. Maybe a teams call will help clear things out.

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring.component.ts (hunk 5 above)
Comment (puneetg1983) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring.component.ts:182]: I had this in my mind but totally skipped it. Thanks, will add some logging.

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html (hunk 6 above)
Comment (puneetg1983) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html:1]: This was a copy paste of the existing collapsible and we fixed accessibility over there so this one just got for free :-)

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html (hunk 3 above)
Comment (puneetg1983) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/crash-monitoring/crash-monitoring-analysis/crash-monitoring-analysis.component.html:21]: Changed to this, looks a bit ugly though. Let me know if you come up with something else

image
